models/record.go

```diff
@@ -591,15 +617,15 @@ func CanonicalizeTargets(recs []*RecordConfig, origin string) {
 		switch r.Type { // #rtype_variations
 		case "AKAMAICDN", "ANAME", "CNAME", "DS", "MX", "NS", "NAPTR", "PTR", "SRV":
 			// Target is a hostname that might be a shortname. Turn it into a FQDN.
-			r.target = dnsutil.AddOrigin(r.target, origin)
+			r.target = dns.Fqdn(dnsutil.AddOrigin(r.target, origin))
```

Comment:
```suggestion
			r.target =dnsutil.AddOrigin(r.target, origin+"."))
```

Reply:
Suggestion: Earlier in this function do: `originDot := origin + "."` then use originDot here and on line 625 and 628.

Reply:
Good one, I added the suggestion (roughly, I like `dns.Fqdn` better to ensure we are not adding a dot after an already , made fqdn.
I also added doc comments to the main function of the dnssort and dnstree. 

Of course I am open to discussing radical changes. If these suggested changes are outside of the re-ordering scope, we can always create a seperate PR for it to not incorporate to much changes in this one. What did you have in mind?